st2actions/st2actions/runners/cloudslang/cloudslang_runner.py

```diff
@@ -0,0 +1,146 @@
+from st2common.constants.action import LIVEACTION_STATUS_FAILED
+from st2common.constants.runners import LOCAL_RUNNER_DEFAULT_ACTION_TIMEOUT
+import st2common.util.jsonify as jsonify
+
+
+LOG = logging.getLogger(__name__)
+
+# constants to lookup in runner_parameters.
+RUNNER_PATH = 'flow_path'
+RUNNER_INPUTS = 'inputs'
+RUNNER_TIMEOUT = 'timeout'
+
+
+def get_runner():
+    return CloudSlangRunner(str(uuid.uuid4()))
+
+
+class CloudSlangRunner(ActionRunner, ShellRunnerMixin):
+    """
+    Runner which executes cloudslang flows and operations as single action
+    """
+    KEYS_TO_TRANSFORM = ['stdout', 'stderr']
+
+    def __init__(self, runner_id):
+        super(CloudSlangRunner, self).__init__(runner_id=runner_id)
+
+    def pre_run(self):
+        self._user = cfg.CONF.system_user.user
+        self._cloudslang_home = cfg.CONF.cloudslang.home_dir
+        self._path = self.runner_parameters.get(RUNNER_PATH)
+        self._inputs = self.runner_parameters.get(RUNNER_INPUTS)
+        self._timeout = self.runner_parameters.get(RUNNER_TIMEOUT,
+                                                   LOCAL_RUNNER_DEFAULT_ACTION_TIMEOUT)
+
+    def run(self, action_parameters):
+        LOG.debug('    action_parameters = %s', action_parameters)
+
+        inputs_file = tempfile.NamedTemporaryFile()
+        has_inputs = self._inputs is not None
+        if has_inputs:
+            LOG.info(self._inputs)
+            inputs_dict = dict(pair.split("=") for pair in self._inputs.split(","))
+            LOG.info(inputs_dict)
+            import yaml
+            inputs_file.write(yaml.safe_dump(inputs_dict, default_flow_style=False))
+            inputs_file.seek(0)
+
+            for line in inputs_file:
+                LOG.info(line.rstrip())
+
+        command = self._cloudslang_home + "/bin/cslang" \
+                                          " run" \
+                                          " --f " + self._path + \
+                                          " --if " + inputs_file.name if has_inputs else "" + \
+                                          " --cp " + self._cloudslang_home
+        action = ShellCommandAction(name=self.action_name,
+                                    action_exec_id=str(self.liveaction_id),
+                                    command=command,
+                                    user=self._user)
+        args = action.get_full_command_string()
+
+        env = os.environ.copy()
+
+        LOG.info('Executing action via LocalRunner: %s', self.runner_id)
+        LOG.info('[Action info] name: %s, Id: %s, command: %s, user: %s, sudo: %s' %
+                 (action.name, action.action_exec_id, args, action.user, action.sudo))
+
+        # Make sure os.setsid is called on each spawned process so that all processes
+        # are in the same group.
+        LOG.info("args are " + args)
+        process = subprocess.Popen(args=args, stdin=None, stdout=subprocess.PIPE,
```

Comment:
To avoid duplicated code, you can use `st2common.util.green.shell.run_command`.

Right now that function doesn't support `preexec_fn` argument and a custom kill command, but you can ignore that for now - I will add support for that later and update the affected code.